packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -909,48 +908,41 @@ export abstract class Chart extends Observable implements AgChartInstance {
 
     private async updateLegend() {
         this.legends.forEach((legend, legendType) => {
-            const legendData: ChartLegendDatum[] = [];
-            this.series
-                .filter((s) => s.showInLegend)
-                .forEach((series) => {
-                    const data = series.getLegendData(legendType);
-                    legendData.push(...data);
-                });
-
-            if (legendType === 'category') {
-                this.validateLegendData(legendData);
+            const isCategory = (_data: unknown): _data is CategoryLegendDatum[] => legendType === 'category';
```

Comment:
Nit: I'd rather we cast using `as` if we're going this route, since it's clearer that we're making assumptions about the type?

Could we just add a real type-guard that actually checks the element types somewhere?

Reply:
we could if we look at the data `legendType`
is this the direction you prefer?

Reply:
Great - thank you!